### PR TITLE
fix: Bumps Flask Caching to fix RCE vulnerability

### LINKED ISSUE #25077: Flask-Caching version 1.10.1, having a critical vulnerability, upgrade to new version is causing a Error.
#### How to reproduce the bug

1. Flask-Caching current version 1.10.1 in Superset 2.1.0
2. Flask-Caching current version 1.10.1 have critical vulnerability.
3. Upgrade Flask-Caching to new version, 1.11.0.
4. than run Superset 2.1.0,
5. Below error will occur in logs.

image

### Expected results

After upgrade Flask-Caching to version 1.11.0, error should not occur

### Actual results

Finding a error in logs

#### Screenshots

image

### Environment

(please complete the following information):

- superset version: 2.1.0
- python version: 3.8.0

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [ ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [ ] I have reproduced the issue with at least the latest released version of superset.
- [ ] I have checked the issue tracker for the same issue and I haven't found one similar.

### Additional context

Flask-Caching current version is 1.10.1 in superset 2.1.0, it is having critical vulnerability (CVE-2021-33026), Aquasec is suggesting to upgrade Flask-Caching to version to 1.11.0
But when we upgraded Flask-Caching to version 1.11.0, we got a error in logs.

Please let me know, how to resolve this Flash-Caching error, so that superset should run smoothly.

Thanks in advance
Nagaraja M M

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
This PR bumps [Flask Caching](https://flask-caching.readthedocs.io/) from `1.10.1` to `1.11.1` to fix [CVE-2021-33026](https://nvd.nist.gov/vuln/detail/CVE-2021-33026). It also removes an explicit dependency with `cachelib` given that its assets are provided via `flask_caching.backends`.

Fixes https://github.com/apache/superset/issues/25077

Here's the [changelog](https://flask-caching.readthedocs.io/en/latest/changelog.html#version-1-11-1) between versions:

Screenshot 2023-08-25 at 17 11 45

### TESTING INSTRUCTIONS
CI should be sufficient.

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by villebro — APPROVED** [review]
LGTM, thanks for all the work on debugging that flaky test, really great work! 👏 Also, please add a note in the description about removing `cachelib` as an explicit dependency, as it may not be obvious why that's happening.

File: requirements/base.txt [hunk 1]
```diff
@@ -31,8 +31,8 @@ bottleneck==1.3.7
     # via pandas
 brotli==1.0.9
     # via flask-compress
-cachelib==0.4.1
-    # via apache-superset
+cachelib==0.6.0
+    # via flask-caching
```
[the remark this answers is not in the document]
Comment (michael-s-molina) [reply requirements/base.txt:35]: Yes, it's safer to do that when bumping Flask-Caching 👍🏼

**Comment by michael-s-molina** [comment]
> Also, please add a note in the description about removing cachelib as an explicit dependency, as it may not be obvious why that's happening.

@villebro I updated the PR description.

**Comment by SCH227** [comment]
@michael-s-molina Flask-caching 1.11.1 does not fixes [CVE-2021-33026](https://nvd.nist.gov/vuln/detail/CVE-2021-33026).
Lastest version of flask-caching still relies on pickle by default for deserialization from cache storage. For example:
[Memcache](https://github.com/pallets-eco/flask-caching/blob/0cb31a3f7b89ce8619b78dfd7597eadb61165229/src/flask_caching/backends/memcache.py#L184)
[Rediscache](https://github.com/pallets-eco/flask-caching/blob/0cb31a3f7b89ce8619b78dfd7597eadb61165229/src/flask_caching/backends/rediscache.py#L104)
Am I missing something?

**Comment by michael-s-molina** [comment]
@SCH227 It looks like the [CVE](https://nvd.nist.gov/vuln/detail/CVE-2021-33026) is still in DISPUTED state pending additional verification. Check https://github.com/pallets-eco/flask-caching/issues/345 for more context.

**Pull request merged**